Assets/MixedRealityToolkit/Utilities/Gltf/Serialization/ConstructGltf.cs

```diff
@@ -0,0 +1,576 @@
+            }
+
+            if (node.children != null)
+            {
+                for (int i = 0; i < node.children.Length; i++)
+                {
+                    await ConstructNodeAsync(gltfObject, gltfObject.nodes[node.children[i]], node.children[i], nodeGameObject.transform);
+                }
+            }
+
+            nodeGameObject.transform.SetParent(parent, false);
+            nodeGameObject.SetActive(true);
+        }
+
+        private static async Task ConstructMeshAsync(GltfObject gltfObject, GameObject parent, int meshId)
+        {
+            GltfMesh gltfMesh = gltfObject.meshes[meshId];
+
+            var renderer = parent.gameObject.AddComponent<MeshRenderer>();
+            var filter = parent.gameObject.AddComponent<MeshFilter>();
+
+            if (gltfMesh.primitives.Length == 1)
+            {
+                gltfMesh.Mesh = await ConstructMeshPrimitiveAsync(gltfObject, gltfMesh.primitives[0]);
+                gltfMesh.Mesh.name = gltfMesh.name;
+                filter.sharedMesh = gltfMesh.Mesh;
+                renderer.sharedMaterial = gltfObject.materials[gltfMesh.primitives[0].material].Material;
+                return;
+            }
+
+            var materials = new List<Material>();
+            var meshCombines = new CombineInstance[gltfMesh.primitives.Length];
+
+            for (int i = 0; i < gltfMesh.primitives.Length; i++)
+            {
+                meshCombines[i].mesh = await ConstructMeshPrimitiveAsync(gltfObject, gltfMesh.primitives[i]);
+
+                var meshMaterial = gltfObject.materials[gltfMesh.primitives[i].material].Material;
+
+                if (!materials.Contains(meshMaterial))
+                {
+                    materials.Add(meshMaterial);
+                }
+            }
+
+            filter.sharedMesh = new Mesh();
+            filter.sharedMesh.CombineMeshes(meshCombines);
+            gltfMesh.Mesh = filter.sharedMesh;
+            gltfMesh.Mesh.name = gltfMesh.name;
+            renderer.sharedMaterials = materials.ToArray();
+        }
+
+        private static async Task<Mesh> ConstructMeshPrimitiveAsync(GltfObject gltfObject, GltfMeshPrimitive meshPrimitive)
+        {
+            if (gltfObject.LoadAsynchronously) await BackgroundThread;
+
+            GltfAccessor positionAccessor = null;
+            GltfAccessor normalsAccessor = null;
+            GltfAccessor textCoord0Accessor = null;
+            GltfAccessor textCoord1Accessor = null;
+            GltfAccessor textCoord2Accessor = null;
+            GltfAccessor textCoord3Accessor = null;
+            GltfAccessor colorAccessor = null;
+            GltfAccessor indicesAccessor = null;
+            GltfAccessor tangentAccessor = null;
+            GltfAccessor weight0Accessor = null;
+            GltfAccessor joint0Accessor = null;
+            int vertexCount = 0;
+
+            if (meshPrimitive.Attributes.POSITION >= 0)
```

Comment:
it feels like we could iterate over meshPrimitives.Attributes values and stub out whatever custom items are needed (like vertex count case by case). right now whats going on below is legible but makes this function super long

Reply:
may need some sort of dictionary accessor for the later assignment

Reply:
Considering that Attributes seems to be a direct deserialized class from the json payload, i can see why this code looks this way/why its a bit harder to change. We may as well keep this this way for now/disregard my suggestions.